app/models/comment.rb

```diff
@@ -49,14 +50,26 @@ class Comment < ActiveRecord::Base
     participation.unparticipate! if participation.present?
   end
 
-  def message
-    @message ||= Diaspora::MessageRenderer.new text
-  end
-
   def text= text
      self[:text] = text.to_s.strip #to_s if for nil, for whatever reason
   end
 
+  def people_allowed_to_be_mentioned
+    if parent.public?
+      :all
+    else
+      [*parent.comments.pluck(:author_id), *parent.likes.pluck(:author_id), parent.author_id].uniq
+    end
+  end
+
+  def subscribers
+    super.tap {|subscribers|
+      if parent.public? && parent.author.local?
+        subscribers.concat(legitimate_mention_recipients.select(&:remote?))
```

Comment:
This is now somehow redundant. You check `public?` here and also check public in `legitimate_mention_recipients` (which calls people_allowed_to_be_mentioned). Also we never need to add subscribers for private posts (status messages and comments)

I would rename `legitimate_mention_recipients` to `add_mention_recipients`, then you can do something like this:
```ruby
    def add_mention_recipients(subscribers)
      subscribers.concat(mentions.map(&:person).select(&:remote?)) if public?
    end
```

And you can use this for comments and status messages.

Just some quick thoughts, maybe you have a better idea how to simplify this? :)